### PR TITLE
tache4.3_NinaVuVan

### PR DESCRIPTION
**Pull request by ninavvn13**

### CODE REVIEW

File: scripts/dimensionality_reduction.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    Parameters:
+    df (pd.DataFrame): The input DataFrame.
+    numeric_cols (list): List of numeric column names.
+    n_components (int): Number of principal components to keep.
+
+    Returns:
+    pd.DataFrame: DataFrame with reduced dimensions.
+    """
+    scaler = StandardScaler()
```
Comment (HerySon) [inline scripts/dimensionality_reduction.py:22]: Laisser la possibilité à l'utilisateur de changer cette méthode de scaling, en passant par exemple le scaler en paramètre

File: scripts/dimensionality_reduction.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    df_scaled = scaler.fit_transform(df[numeric_cols])
```
Comment (HerySon) [inline scripts/dimensionality_reduction.py:23]: Encore une fois ici, pour des raisons de mémoire vive, tu peux passer éviter d'enregistrer ce traitement dans une nouvelle variable, mais plutôt écraser `df`

File: scripts/dimensionality_reduction.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    dict: Best parameters found by GridSearchCV.
+    """
+    X = df[numeric_cols]
+    y = df[target_col]
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
+
+    pipeline = Pipeline([
+        ('scaler', StandardScaler()),
+        ('pca', PCA()),
+        ('classifier', LogisticRegression(max_iter=1000))
```
Comment (HerySon) [inline scripts/dimensionality_reduction.py:56]: Attention, ici il n'est pas demandé d'optimiser une Régression Logisitique, cette partie du code est hors sujet. L'optimisation de la PCA peut être intéressante, mais elle tu ne pourras pas la faire avec un **GridSearch car PCA est un algorithme non supervisé et donc n'a pas de label**

File: scripts/dimensionality_reduction.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    grid_search = GridSearchCV(pipeline, param_grid, cv=5, scoring='accuracy')
+    grid_search.fit(X_train, y_train)
+
+    print(f"Best parameters: {grid_search.best_params_}")
+    print(f"Best accuracy: {grid_search.best_score_}")
+
+    return grid_search.best_params_
+
+def select_features(df, numeric_cols, n_components=2):
```
Comment (HerySon) [inline scripts/dimensionality_reduction.py:67]: Je ne comprends pas quel est l'intérêt de cette fonction sachant que tu répètes le code de ta fonction pca_reduction ? Je m'attendais plutôt à une fonction qui te permette de filtrer les composantes à ne pas garder sur la base du pourcentage de variance expliquée cumulée

**Review by HerySon — CHANGES_REQUESTED** [review]

**Comment by HerySon** [comment]
@ninavvn13
